src/main.tf

```diff
@@ -1,3 +1,8 @@
+locals {
+  node_rg_max_length = 80
+  node_rg_name       = substr("MC_${var.md_metadata.name_prefix}_${var.md_metadata.name_prefix}_${var.vnet.specs.azure.region}", 0, local.node_rg_max_length)
```

Comment:
Do the nodes go in their own resource group or something? Is that why it auto-gens this? It seems really strange to have the cluster and nodes in separate resource groups.

Also, what happens if the resource group name is >80. Is that allowed, or does it get truncated? If it's allowed, how do we reference it when the field limit here is 80?

It seems like there is a bigger issue here. I'm baffled by the custom auto-generated resource group.

Reply:
Correct. It must create a fresh, unused resource group and also be the only entity that uses it.

If the resource group name is >80, it fails to deploy. It doesn't get truncated or anything like that. Simply fails to deploy.

Reply:
![image](https://github.com/massdriver-cloud/azure-aks-cluster/assets/8037512/262eb8cc-bd78-4f16-a4a4-11d8e9cf2b25)
![image](https://github.com/massdriver-cloud/azure-aks-cluster/assets/8037512/2eea6f58-e683-4a70-9033-891edb0b6bbd)
@chrisghill https://learn.microsoft.com/en-us/azure/aks/faq#why-are-two-resource-groups-created-with-aks
> You create the first resource group. This group contains only the Kubernetes service resource. The AKS resource provider automatically creates the second resource group during deployment. An example of the second resource group is MC_myResourceGroup_myAKSCluster_eastus. For information on how to specify the name of this second resource group, see the next section.
> The second resource group, known as the node resource group, contains all of the infrastructure resources associated with the cluster. These resources include the Kubernetes node VMs, virtual networking, and storage. By default, the node resource group has a name like MC_myResourceGroup_myAKSCluster_eastus. AKS automatically deletes the node resource group whenever you delete the cluster. You should only use this cluster for resources that share the cluster's lifecycle.